besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java

```diff
@@ -988,6 +988,20 @@ public void genesisAndNetworkMustNotBeUsedTogether() throws Exception {
         .startsWith("--network option and --genesis-file option can't be used at the same time.");
   }
 
+  @Test
+  public void testDnsDiscoveryUrlEthConfig() throws Exception {
+    final ArgumentCaptor<EthNetworkConfig> networkArg =
+            ArgumentCaptor.forClass(EthNetworkConfig.class);
+
+    parseCommand("--dns-discovery-url", "test123");
```

Comment:
Testing with fake/wrong values is fine, but you should also respect the value syntax: `enrtree://<key>@<fqdn>` as defined in [eip-1459](https://eips.ethereum.org/EIPS/eip-1459).
It will not test that the DNS query works, it's not in the scope of the test here (and mocked), but it helps making sure it at least works with the method you are testing. Plus it helps making the tests understandable for readers. Don't forget that tests are used to test... obviously... but also to understand how things works. Tests are the first doc after code itself. We look at test very frequently to write the doc and create examples.